app/decorators/notification_decorator.rb

```diff
@@ -45,6 +45,49 @@ def to_model
     self
   end
 
+  def subscription_for(user)
+    subscription_to_comment_ancestry_for(user) ||
+      subscription_to_comment_for(user) ||
+      subscription_to_commentable_article_for(user) ||
+      subscription_to_article_for(user)
+  end
+
+  def subscription_to_article_for(user)
+    return unless article_id
+
+    subscription_to_notifiable_for(user,
+                                   notifiable_type: "Article",
+                                   notifiable_id: article_id)
+  end
+
+  def subscription_to_comment_for(user)
+    return unless comment_ancestry
+
+    subscription_to_notifiable_for(user,
+                                   notifiable_type: "Comment",
+                                   notifiable_id: comment_id)
+  end
+
+  def subscription_to_comment_ancestry_for(user)
+    return unless comment_ancestry
+
+    subscription_to_notifiable_for(user,
+                                   notifiable_type: "Comment",
+                                   notifiable_id: comment_ancestry.split("/"))
+  end
+
+  def subscription_to_commentable_article_for(user)
```

Comment:
A _potential_ optimization if we _do_ see performance issues might be to try combining the two comment queries into one.

In either case, a risk here is that if a user has subscriptions at multiple places in the same thread (as they likely will if they are the author of various [grand]parent comments) we don't currently guarantee that we find the "closest subscription". (Sorting subscriptions by `notifiable_id` might work, if comment ids are guaranteed to be incrementing...)